edge-functions/breakpoint-server/pages/[breakpoint]/index.tsx

```diff
@@ -0,0 +1,49 @@
+import type { GetStaticPaths, GetStaticProps } from 'next'
+import Head from 'next/head'
+import type { ParsedUrlQuery } from 'querystring'
+import { Layout } from '@vercel/examples-ui'
+import { Breakpoint, BreakpointServer, serverBreakpoints } from '../../config/breakpoints'
+import { useBreakpoint } from '../../hooks/use-breakpoint/use-breakpoint'
+
+interface Params extends ParsedUrlQuery {
+  breakpoint: BreakpointServer
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const paths = serverBreakpoints.map((breakpoint) => ({
+    params: { breakpoint },
+  }))
+
+  return {
+    paths: paths,
+    fallback: true, // fallback true allows sites to be generated using ISR
+  }
+}
+
+export const getStaticProps: GetStaticProps<unknown, Params> = async ({
+  params: { breakpoint },
+}) => {
+  return {
+    props: {
+      breakpoint,
+    },
+    revalidate: 60,
+  }
+}
+
+export default function HomePage() {
+  const size = useBreakpoint()
+  const isSmall = size === Breakpoint.xs || size === Breakpoint.sm
```

Comment:
Am I right in thinking that the breakpoint is actually going to be decided by the `getBreakpoint` function inside `useEffect`? Even though that works, I would like to see it being decided by the edge instead, otherwise it's recommended to use CSS to avoid a CLS of content in the browser.

Reply:
But @lfades thats actually the case, see https://github.com/vercel/examples/blob/f1d50db0c420e69d21b7f46db833fe53a31585e1/edge-functions/breakpoint-server/pages/_middleware.ts#L34 in the middleware the initialBreakpoint is passed to the application at https://github.com/vercel/examples/blob/f1d50db0c420e69d21b7f46db833fe53a31585e1/edge-functions/breakpoint-server/pages/_app.tsx#L20 

So server-side its in the breakpoint we got on the edge. Then after that client-side kicks in and updates it to the real value. Since user-agent trick is just to get a better fit.